ansible/mnesia_snapshot.yml

```diff
@@ -21,39 +27,51 @@
 
     - meta: flush_handlers
 
-    - name: Archive Mnesia Directory
+    - name: Ensure Mnesia snapshots dir exists ({{ snapshots_dir }})
+      file:
+        path: "{{ snapshots_dir }}"
+        state: directory
+
+    - name: Archive Mnesia Directory ({{ project_root }}/{{ node_config.chain.db_path }})
+      archive:
+        format: gz
+        path: "{{ project_root }}/{{ node_config.chain.db_path }}"
+        dest: "{{ snapshot_path }}"
+      when: additional_storage is not defined or not addtional_storage
+
+    - name: Archive Mnesia Directory ({{ node_config.chain.db_path }})
       archive:
-        format: "{{ format }}"
-        path: "{{ project_root + '/' if additional_storage is not defined else '' }}{{ node_config.chain.db_path }}"
-        dest: /tmp/{{ backup_archive_filename }}
+        format: gz
+        path: "{{ node_config.chain.db_path }}"
+        dest: "{{ snapshot_path }}"
+      when:
+        - additional_storage is defined
+        - additional_storage
 
-    - name: Upload backup to S3 bucket
+    - name: Upload snapshot to S3 bucket ({{ snapshots_bucket }}/{{ snapshot_filename }})
       aws_s3:
-        bucket: "{{ backup_bucket }}"
-        object: "{{ backup_archive_filename }}"
-        src: /tmp/{{ backup_archive_filename }}
+        bucket: "{{ snapshots_bucket }}"
+        object: "{{ snapshot_filename }}"
+        src: "{{ snapshot_path }}"
         mode: put
         aws_access_key: "{{ lookup('env','AWS_ACCESS_KEY_ID') }}"
         aws_secret_key: "{{ lookup('env','AWS_SECRET_ACCESS_KEY') }}"
         security_token: "{{ lookup('env','AWS_SESSION_TOKEN') }}"
 
     - name: Upload backup to S3 bucket latest
       aws_s3:
-        bucket: "{{ backup_bucket }}"
-        object: "{{ backup_archive_filename_latest }}"
-        src: /tmp/{{ backup_archive_filename }}
+        bucket: "{{ snapshots_bucket }}"
+        object: "{{ snapshots_filename_latest }}"
+        src: "{{ snapshot_path }}"
         mode: put
         aws_access_key: "{{ lookup('env','AWS_ACCESS_KEY_ID') }}"
         aws_secret_key: "{{ lookup('env','AWS_SECRET_ACCESS_KEY') }}"
         security_token: "{{ lookup('env','AWS_SESSION_TOKEN') }}"
 
-    - name: Remove Mnesia backups
+    - name: Delete local Mnesia snapshots ({{ snapshots_dir }})
       file:
-        path:  "{{ item }}"
+        path: "{{ snapshots_dir }}"
```

Comment:
All? or snapshot_path?

Reply:
Yes, it is meant to delete all snapshots in that directory. The use-case is if something fail in the middle, next time it's cleanedup.